nomt/src/bitbox/beatree/allocator/mod.rs

```diff
@@ -176,8 +179,7 @@ impl AllocatorWriter {
             free_list_pages,
             bump: self.bump,
             extend_file_sz,
-            // after appending to the free list, the head will always be present
-            freelist_head: self.free_list.head_pn().unwrap(),
+            freelist_head: self.free_list.head_pn().unwrap_or(FREELIST_EMPTY),
```

Comment:
yes, thanks for the correction! It can be empty in the case of all free pages being released